shardingsphere-control-panel/shardingsphere-orchestration/shardingsphere-orchestration-core/shardingsphere-orchestration-core-registrycenter/src/main/java/org/apache/shardingsphere/orchestration/core/registrycenter/instance/OrchestrationInstance.java

```diff
@@ -29,21 +28,30 @@
 public final class OrchestrationInstance {
     
     private static final String DELIMITER = "@";
-    
-    private static final OrchestrationInstance INSTANCE = new OrchestrationInstance();
+
+    private static OrchestrationInstance instance;
     
     private final String instanceId;
-    
-    private OrchestrationInstance() {
-        instanceId = IpUtils.getIp() + DELIMITER + ManagementFactory.getRuntimeMXBean().getName().split(DELIMITER)[0] + DELIMITER + UUID.randomUUID().toString();
+
+    public OrchestrationInstance(final int identifier) {
+        instanceId = IpUtils.getIp() + DELIMITER + identifier + DELIMITER + UUID.randomUUID().toString();
```

Comment:
How about add `instance = this;`

Reply:
what we really want is a singleton instance with the property accessible everywhere.
probably we should keep the constructor as it is and provide a `getInstance()` method throwing `IllegalStateException` if instance object is not created

I do not think making an assignment like `instance = this` is a good idea.
@menghaoranss thoughts?

Reply:
> what we really want is a singleton instance with the property accessible everywhere.
> probably we should keep the constructor as it is and provide a `getInstance()` method throwing `IllegalStateException` if instance object is not created
> 
> I do not think making an assignment like `instance = this` is a good idea.
> @menghaoranss thoughts?

Agree with you, may be we can keep the no arguments constructor , and then add a new method to generate `instanceId`. Generate `instanceId` when `OrchestrationInstance` is initialized in `RegistryCenter`.

Reply:
@menghaoranss 
i am thinking of keeping the impl like this,
```
public static OrchestrationInstance init(final String instanceTag) {
        if (null == instance) {
            synchronized (OrchestrationInstance.class) {
                if (null == instance) {
                    instance = new OrchestrationInstance(instanceTag);
                }
            }
        }
        return instance;
    }

    public static String getInstanceId() {
        if (null == instance) {
            throw new IllegalStateException("OrchestrationInstance not initialized!");
        }
        return instance.instanceId;
    }
```
but because of this code `InstanceStateChangedListenerTest.setUp()` throws the `IllegalStateException`.
Will need to refactor the test-case as well.

Should this be ok?

Reply:
I can understand the meaning of your code above, but I think it is too complicated, below is my suggestion: 

```
@Getter
public final class OrchestrationInstance {
    
    private static final String DELIMITER = "@";
    
    private static final OrchestrationInstance INSTANCE = new OrchestrationInstance();
    
    private String instanceId;
    
    private OrchestrationInstance() {
    
    }
    
    public void init(final String instanceTag) {
        instanceId = IpUtils.getIp() + DELIMITER + instanceTag + DELIMITER + UUID.randomUUID().toString();
    }
    
    /**
     * Get instance.
     * 
     * @return instance
     */
    public static OrchestrationInstance getInstance() {
        return INSTANCE;
    }
}
```

Reply:
@menghaoranss  If we go ahead with this approach, we will risk someone calling `getInstance()` without ensuring `init()` was called and then returning an empty string as instanceId.
Moreover, this makes instanceId mutable i.e. anyone can do `OrchestrationInstance.getInstance().init("a")` and then `OrchestrationInstance.getInstance().init("b")`

Reply:
> @menghaoranss If we go ahead with this approach, we will risk someone calling `getInstance()` without ensuring `init()` was called and then returning an empty string as instanceId.
> Moreover, this makes instanceId mutable i.e. anyone can do `OrchestrationInstance.getInstance().init("a")` and then `OrchestrationInstance.getInstance().init("b")`

Yes, we must guarantee that `OrchestrationInstance` only initialize once in the code，this is what we need to do.

Reply:
> @menghaoranss If we go ahead with this approach, we will risk someone calling `getInstance()` without ensuring `init()` was called and then returning an empty string as instanceId.
> Moreover, this makes instanceId mutable i.e. anyone can do `OrchestrationInstance.getInstance().init("a")` and then `OrchestrationInstance.getInstance().init("b")`

Could you perfect this pr?

Reply:
I have committed new changes, can you please review that.

Reply:
I have reviewed your last submission, and the points that we discussed above still need to be optimized.